tools/distpartitioning/data_shuffle.py

```diff
@@ -142,6 +146,27 @@ def gen_node_data(
 
         for local_part_id in range(num_parts // world_size):
             cond = node_partid_slice == (rank + local_part_id * world_size)
+
+        # No. of nodes in each process can differ significantly in lopsided distributions
+        # Synchronize on a per ntype basis
+        dist.barrier()
+
+        type_start, type_end = (
+            type_nid_dict[ntype_name][0][0],
+            type_nid_dict[ntype_name][-1][1],
+        )
+        gnid_start, gnid_end = (
+            global_nid_dict[ntype_name][0, 0],
+            global_nid_dict[ntype_name][0, 1],
+        )
+
+        node_partid_slice = id_lookup.get_partition_ids(
+            np.arange(gnid_start, gnid_end, dtype=np.int64)
+        )  # exclusive
+
+        for local_part_id in range(num_parts // world_size):
```

Comment:
duplicate to line 149~150. seem previous ones should be removed.

Reply:
Lot of repeated/duplicate code was introduced after the merge conflict resolutions were made. 
It appears that the merge conflicts are very tricky and github fails more often then we thought. 

Now cleaned up all the issues, which were introduced because of the merge conflicts.